pkg/healthcheck/healthcheck.go

```diff
@@ -694,6 +709,125 @@ func (hc *HealthChecker) allCategories() []category {
+					hintAnchor:  "l5d-crt-config-is-valid",
+					fatal:       true,
+					check: func(context.Context) (err error) {
+						hc.issuerCert, hc.roots, err = hc.checkCertificatesConfig()
+						return
+					},
+				},
+				{
+					description: "control plane is using supported trust roots",
+					hintAnchor:  "l5d-supported-certs-type",
+					fatal:       true,
+					check: func(context.Context) error {
+						var invalidRoots []string
+						for _, root := range hc.roots {
+							if err := issuercerts.CheckCertAlgoRequirements(root); err != nil {
+								invalidRoots = append(invalidRoots, fmt.Sprintf("* %v %s %s", root.SerialNumber, root.Subject.CommonName, err))
+							}
+						}
+						if len(invalidRoots) > 0 {
+							return fmt.Errorf("Invalid roots:\n\t%s", strings.Join(invalidRoots, "\n\t"))
+						}
+						return nil
+					},
+				},
+				{
+					description: "identity is using supported issuer certificate",
+					hintAnchor:  "l5d-supported-certs-type",
+					fatal:       true,
+					check: func(context.Context) error {
+						if err := issuercerts.CheckCertAlgoRequirements(hc.issuerCert.Certificate); err != nil {
+							return fmt.Errorf("issuer certificate %s", err)
+						}
+						return nil
+					},
+				},
+				{
+					description: "all control plane trust roots have valid lifetimes",
+					hintAnchor:  "l5d-certs-rotation",
+					fatal:       true,
+					check: func(ctx context.Context) error {
+						var expiredRoots []string
+						for _, root := range hc.roots {
+							if err := issuercerts.CheckCertTimeValidity(root); err != nil {
+								expiredRoots = append(expiredRoots, fmt.Sprintf("* %v %s %s", root.SerialNumber, root.Subject.CommonName, err))
+							}
+						}
+						if len(expiredRoots) > 0 {
+							return fmt.Errorf("Invalid roots:\n\t%s", strings.Join(expiredRoots, "\n\t"))
+						}
+
+						return nil
+					},
+				},
+				{
+					description: "no control plane trust roots are near expiry",
+					hintAnchor:  "l5d-certs-rotation",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						var expiringRoots []string
+						for _, root := range hc.roots {
+							if err := issuercerts.CheckExpiringSoon(root); err != nil {
+								expiringRoots = append(expiringRoots, fmt.Sprintf("* %v %s %s", root.SerialNumber, root.Subject.CommonName, err))
+							}
+						}
+						if len(expiringRoots) > 0 {
+							return fmt.Errorf("Roots expiring soon:\n\t%s", strings.Join(expiringRoots, "\n\t"))
+						}
+						return nil
+					},
+				},
+				{
+					description: "issuer cert has valid lifetime",
+					hintAnchor:  "l5d-certs-rotation",
+					fatal:       true,
+					check: func(ctx context.Context) error {
+						if err := issuercerts.CheckCertTimeValidity(hc.issuerCert.Certificate); err != nil {
+							return fmt.Errorf("issuer certificate is %s", err)
+						}
+						return nil
+					},
+				},
+				{
+					description: "issuer cert is not near expiry",
```

Comment:
Nit: I think that all checks should be stated in the affirmative.  For example: "issuer cert is valid for at least 60 days"

When these checks fail, I find it hard to mentally parse the double-negative